addons/mail/wizard/mail_compose_message.py

```diff
@@ -701,18 +704,33 @@ def _action_send_mail_mass_mail(self, res_ids, auto_commit=False):
         return mails_sudo
 
     def action_save_as_template(self):
-        """ hit save as template button: current form value will be a new
-            template attached to the current document. """
+        """ hit save as template button: opens a wizard that prompts for the template's subject.
+            `save_as_template` is called when saving the new wizard. """
+
+        wizard = self.copy()
+        wizard.subject = False
+        return {
+            'type': 'ir.actions.act_window',
+            'view_mode': 'form',
+            'view_id': self.env.ref('mail.email_compose_message_wizard_small_form').id,
+            'name': 'Create a new Template',
+            'res_model': 'mail.compose.message',
+            'context': {'dialog_size': 'medium'},
+            'target': 'new',
+            'res_id': wizard.id,
+        }
+
+    def save_as_template(self):
```

Comment:
Would make a private method, like
```suggestion
    def _create_mail_template(self):
```

Reply:
The method is called from a button from the small wizard so it can't be private

Reply:
Maybe at least update naming to be clearer ?